packages/@tinacms/cli/src/cmds/start-server/index.ts

```diff
@@ -192,6 +192,8 @@ export async function startServer(
   let isReady = false
 
   const start = async () => {
+    // we do not want to start the server while the schema is building
+    await lock.promise
```

Comment:
is there a risk that the lock will be taken right after this? should this hold the lock until it starts?

Reply:
I don't think there is a risk of this since the schema compiling always starts before the server.